configs/components/ruby-2.7.1.rb

```diff
@@ -0,0 +1,239 @@
+component 'ruby-2.7.1' do |pkg, settings, platform|
+  pkg.version '2.7.1'
+  pkg.sha256sum 'd418483bdd0000576c1370571121a6eb24582116db0b7bb2005e90e250eae418'
+
+  # rbconfig-update is used to munge rbconfigs after the fact.
+  pkg.add_source("file://resources/files/ruby/rbconfig-update.rb")
+
+  # PDK packages multiple rubies and we need to tweak some settings
+  # if this is not the *primary* ruby.
+  if (pkg.get_version != settings[:ruby_version])
+    # not primary ruby
+
+    # ensure we have config for this ruby
+    unless settings.has_key?(:additional_rubies) && settings[:additional_rubies].has_key?(pkg.get_version)
```

Comment:
```ruby
unless settings[:additional_rubies] && settings[:additional_rubies][pkg.get_version]
```

Reply:
hmm, I feel that `settings.key?(:additional_rubies)` is a more idiomatic way to check if a key exists, rather than directly accessing the key, so I'm gonna change it to that